examples/moving_least_squares/moving_least_squares.cpp

```diff
@@ -0,0 +1,80 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX.hpp>
+#include <ArborX_Interp.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <iostream>
+
+template <typename T>
+KOKKOS_INLINE_FUNCTION double step(T const &p)
+{
+  return Kokkos::signbit(p[0]) ? 0 : 1;
+}
+
+using Point = ArborX::ExperimentalHyperGeometry::Point<1, double>;
+using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+using MemorySpace = typename ExecutionSpace::memory_space;
+
+int main(int argc, char *argv[])
+{
+  Kokkos::ScopeGuard guard(argc, argv);
+  ExecutionSpace space{};
+
+  static constexpr std::size_t num_points = 1000;
+
+  Kokkos::View<Point *, MemorySpace> source_points(
+      Kokkos::view_alloc(Kokkos::WithoutInitializing, "Example::source_points"),
+      num_points);
+  Kokkos::View<double *, MemorySpace> source_values(
+      Kokkos::view_alloc(Kokkos::WithoutInitializing, "Example::source_values"),
+      num_points);
+  Kokkos::View<Point *, MemorySpace> target_points(
+      Kokkos::view_alloc(Kokkos::WithoutInitializing, "Example::target_points"),
+      num_points);
+  Kokkos::View<double *, MemorySpace> target_values(
+      Kokkos::view_alloc(Kokkos::WithoutInitializing, "Example::target_values"),
+      num_points);
+  Kokkos::parallel_for(
```

Comment:
Add a comment to describes how you are placing the points.  Something like "Source points are evenly distributed points along a line between -1 and 1 and target points are offset, centered between source points".

Reply:
I have changed that to an actual random point cloud